### PR TITLE
Extend callbacks to brnn

### PR DESCRIPTION
**Pull request by jeffin143**
Handles #2013 , Extending callbacks to brnn

### CODE REVIEW

File: src/mlpack/tests/recurrent_network_test.cpp [hunk 1]
```diff
@@ -1413,4 +1414,82 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
...
+}
+
+/**
+ * Test a BRNN model with PrintLoss callbacks.
+ */
+BOOST_AUTO_TEST_CASE(BRNNCallbackTest)
+{
+  const size_t rho = 10;
+
+  arma::cube input;
```
Comment (zoq) [inline src/mlpack/tests/recurrent_network_test.cpp]: Strange, looks like the test failed for the same model; I think you used the same model for each callback test?

Comment (jeffin143) [reply src/mlpack/tests/recurrent_network_test.cpp]: Yes , is there any issue , since Between each boost test case destructor are called !

Comment (zoq) [reply src/mlpack/tests/recurrent_network_test.cpp]: Actually I'm not sure, let's see if I can replicate the issue on the Mac setup I have.

Comment (jeffin143) [reply src/mlpack/tests/recurrent_network_test.cpp]: @zoq any update on this , were you able to replicate ?

Comment (zoq) [reply src/mlpack/tests/recurrent_network_test.cpp]: Sorry, haven't had a chance yet to take a look, will put it on my list for tomorrow.

Comment (shrit) [reply src/mlpack/tests/recurrent_network_test.cpp]: @zoq is there anything missing in this pull request, can we help??

Comment (zoq) [reply src/mlpack/tests/recurrent_network_test.cpp]: If you like you can try to reproduce the error in ther macOS build: https://travis-ci.org/mlpack/mlpack/jobs/594423997?utm_medium=notification&utm_source=github_status

**Comment by walragatver** [comment]
Hi @jeffin143,
Can you just rebase this PR? As you know we have azure pipelines for mac os checks. Let's see if it fails there too. If it then we can dig into it.

**Comment by walragatver** [comment]
Hi @jeffin143, 
It looks like it's not failing anymore. So if you can fix those style errors then it will be ready to get merge.

**Comment by jeffin143** [comment]
> Hi @jeffin143,
> It looks like it's not failing anymore. So if you can fix those style errors then it will be ready to get merge.

@walragatver error still persists , not sure why

**Comment by walragatver** [comment]
Okay @jeffin143, I will try to look into it. Do you think we should open an issue for this? It would be good for someone who is trying to go through the codebase. Let me know what you think.

**Comment by jeffin143** [comment]
> Okay @jeffin143, I will try to look into it. Do you think we should open an issue for this? It would be good for someone who is trying to go through the codebase. Let me know what you think.

I guess it would be apt, but I am not sure how to pen down the way out to the issue, could you open an issue, don't feel obliged :)

**Comment by rcurtin** [comment]
This code touches the ANN code, and there was recently a big refactoring of the ANN code in #2259, so be sure to merge the master branch into your branch here to make sure that nothing will fail if this PR is merged. :+1: (I'm pasting this message into all possibly relevant PRs.)

**Comment by jeffin143** [comment]
Last commit https://github.com/mlpack/mlpack/pull/2046/commits/c839af08ac88c1fce672c18246a4cc58bcdc6b7b was in order to resolve #2523 a bit

**Comment by jeffin143** [comment]
keep open

**Comment by rcurtin** [comment]
For what it's worth, after #2777 the BRNN class is not adapted to the new inheritance-based approach, so solving that would be a prerequisite for this too.  It might be a lot of work, I am not sure.

**Comment by jeffin143** [comment]
Will take a look at the EOM

**Pull request closed without merging**